imap_processing/ultra/l1b/ultra_l1b_extended.py

```diff
@@ -258,3 +267,53 @@ def get_path_length(front_position: tuple, back_position: tuple, d: float) -> fl
     )
 
     return r
+
+
+def get_ssd_back_position_and_tof_offset(
+    de_dataset: xarray.Dataset,
+) -> tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """
+    Lookup the Y SSD positions (yb), TOF Offset, and SSD number.
+
+    Parameters
+    ----------
+    de_dataset : xarray.Dataset
+        The input dataset containing STOP_TYPE and SSD_FLAG data.
+
+    Returns
+    -------
+    yb : np.ndarray
+        Y SSD positions in hundredths of a millimeter.
+    tof_offset : np.ndarray
+        TOF offset.
+    ssd_number : np.ndarray
+        SSD number.
+
+    Notes
+    -----
+    The X back position (xb) is assumed to be 0 for SSD.
+    """
+    indices = np.nonzero(np.isin(de_dataset["STOP_TYPE"], StopType.SSD.value))[0]
+    de_filtered = de_dataset.isel(epoch=indices)
+
+    yb = np.zeros(len(indices), dtype=np.float64)
+    ssd_number = np.zeros(len(indices), dtype=int)
+    tof_offset = np.zeros(len(indices), dtype=np.float64)
+
+    for i in range(8):
+        # Multiply ybs times 100 to convert to hundredths of a millimeter.
+        yb[de_filtered[f"SSD_FLAG_{i}"].data == 1] = (
```

Comment:
There are 3 times that this mask `de_filtered[f"SSD_FLAG_{i}"].data == 1` is created. You could store that in a variable to make this a tiny bit more efficient.